manager/allocator/network.go

```diff
@@ -93,22 +93,28 @@ func (a *Allocator) doNetworkInit(ctx context.Context) (err error) {
 		}
 	}()
 
-	// Check if we have the ingress network. If not found create
-	// it before reading all network objects for allocation.
-	var networks []*api.Network
-	a.store.View(func(tx store.ReadTx) {
-		networks, err = store.FindNetworks(tx, store.ByName(ingressNetworkName))
-		if len(networks) > 0 {
-			nc.ingressNetwork = networks[0]
-		}
-	})
-	if err != nil {
-		return errors.Wrap(err, "failed to find ingress network during init")
+	// Retrieve current cluster ID and routing mesh state
+	var clusters []*api.Cluster
+	if a.store.View(func(readTx store.ReadTx) {
+		clusters, err = store.FindClusters(readTx, store.ByName(store.DefaultClusterName))
+	}); err != nil {
+		return errors.Wrapf(err,
+			"failed to retrieve cluster object to check routing mesh state during init: %v", err)
 	}
 
-	// If ingress network is not found, create one right away
-	// using the predefined template.
-	if len(networks) == 0 {
+	nc.clusterID = clusters[0].ID
```

Comment:
You should check that `clusters` has a nonzero length. `FindClusters` will not return an error if it doesn't find any matching clusters.

There should always be a `default` cluster, so it would be a bug if one didn't exist. But I think it's better to return an error than crash in that case.